app/services/umrdr/date_coverage_service.rb

```diff
@@ -0,0 +1,130 @@
+                      integerize(params[:date_coverage_end_year]),
+                      integerize(params[:date_coverage_end_month]),
+                      integerize(params[:date_coverage_end_day])
+        )
+        EDTF::Interval.new(begin_date, end_date) unless begin_date == :unknown && end_date == :open
+      end
+
+      # Given an EDTF Interval
+      # Returns a corresponding hash of parameters for an edit form.
+      def interval_to_params( interv )
+        return unless interv.is_a? EDTF::Interval
+        reverse_interval!(interv) if interval_reversed?(interv) 
+        make_begin_params(interv.begin).merge(make_end_params(interv.end))
+      end
+
+      private
+
+      # Determines if an intervals endings are reverse.  Specifically, where the begin 
+      # date is later that the end date.  An interval with reversed starting and ending
+      # dates will have from and to attributes, but begin and end will be nil
+      # == Parameters:
+      # interv::
+      #   [EDTF::Interval] The interval to be tested. 
+      # == Returns:
+      #   [Boolean] True if the interval boundaries were reversed
+      def interval_reversed?(interv)
+        return false if interv.from == EDTF_OPEN_BEGIN || interv.to == EDTF_OPEN_END 
+        (interv.from && !interv.begin) && (interv.to && !interv.end)
+      end
+
+      # Reverse the begin and end of an interval.  This will bork a well formed interval.
+      # == Parameters:
+      # interv::
+      #   [EDTF::Interval] The interval to be reversed. 
+      # == Returns:
+      #   [EDTF::Interval] Interval with from and to in the reverse order from the input interval.
+      def reverse_interval!(interv)
+        tmp = interv.to
+        interv.to = interv.from
+        interv.from = tmp
+      end
+
+      def make_begin_params(date)
+        hsh = date_to_params date
+        {date_coverage_begin_year:  hsh[:year],
+         date_coverage_begin_month: hsh[:month],
+         date_coverage_begin_day:   hsh[:day]}
+      end
+
+      def make_end_params( date )
+        hsh = date_to_params date
+        {date_coverage_end_year:  hsh[:year],
+         date_coverage_end_month: hsh[:month],
+         date_coverage_end_day:   hsh[:day]}
+      end
+
+      # Given edtf date
+      # return hash with possible keys :year, :month, :day
+      def date_to_params( date )
+        return {} unless date
+        hsh = case date.precision
+        when :year
+          {year: date.year}
+        when :month
+          {year: date.year, month: date.month}
+        when :day
+          {year: date.year, month: date.month, day: date.day}
+        end
+        hsh[:year] = year_to_s hsh[:year]
+        hsh.each{|k,v| hsh[k] = v.to_s}
+      end
+
+      def year_to_s(val)
+        if val.negative?
+          sprintf('-%04d', val.abs)
+        else
+          sprintf('%04d', val)
+        end
+      end
+
+      # Given input
+      # Return the corresponding integer or nil
+      def integerize(input)
```

Comment:
I would call it something like `#safe_to_i` or `#safe_Integer`. This is probably also a method you want to pull out and put into a refinement, module, or throw in `Kernel`. You could then make it a proper method that uses `self` instead of taking an input.